RFCs/FS-1027-fsi-references.md

```diff
@@ -164,6 +162,18 @@ This will generate an error because `System.Whatever` version `4.1.0` will alrea
 
 We may consider using `AssemblyLoadContext` in some clever way in the future, but it's very difficult to get right and well outside the scope of what we need to support for a .NET Core 2.0 timeframe.
 
+## Handler resolution
+
+FSI/Design time support will look at the following places in order:
+
+* if the script is a physical file, check current folder and browse all parent folders looking for .fsharp/fsi-extensions folder in each one
+* look into ~/.fsharp/fsi-extensions
+* look into .fsharp/fsi-extensions folder next to fsi.exe
+
+gather all the distinct dll names, order of precedence favorising those in the same order shown above, and load them in the process if their assembly contains an arbitrary attribute (resolved by name rather than dependency on external library) and types marked with same attributes.
```

Comment:
We need more detail on the loading process here. 

* When you say ``gather all the distinct dll names`` do you look in immediate subdirectories?  I believe we should since each extension could be a whole tool of its own, with its own dependencies and .dl.config binding config?  

* Could you specify that for ``fsi,exe`` on .NET Framework these are loaded in-process using ``Assembly.LoadFrom``. 

* The same will apply to ``devenv.exe`` and other design-time tooling based on ``FSharp.Compiler.Service.dll`` such as anything based on ``FsAutoComplete.exe``.  We should specify whether this means the DLLs are loaded multiple times from different locations as you open and work with different scripts.

* It's possible that in the world of .NET Core this loading will look different.  I don't yet fully understand what the natural notion of an application extension is in the word of .NET Core and if there's a natural notion of a .NET Standard extension.  In any case we should note this and mention that we will need to specify whether extensions can be .NET Standard 1.6 DLLs and if so whether they will be loadable in both .NET Core and .NET Framework compilers